src/components/Graph/Graph.tsx

```diff
@@ -1,184 +1,149 @@
-import React, {
-  FC,
-  RefObject,
-  useCallback,
-  useEffect,
-  useMemo,
-  useRef,
-  useState,
-  memo
-} from 'react';
+import React, { FC, useCallback, useEffect, useMemo, useRef, useState, memo } from 'react';
 import { useParams, useHistory } from 'react-router-dom';
 
 import graphLayout from 'visualization/layout';
 import Node from 'components/Node';
 import Arrow from 'components/Arrow';
-import { useCriteriaContext } from 'components/CriteriaProvider';
+import { useBuildCriteriaContext } from 'components/BuildCriteriaProvider';
 import { Pathway, PathwayNode } from 'pathways-model';
 import { Layout, NodeDimensions, NodeCoordinates, Edges } from 'graph-model';
 import styles from './Graph.module.scss';
-import ResizeSensor from 'css-element-queries/src/ResizeSensor';
+import { useCurrentPathwayContext } from 'components/CurrentPathwayProvider';
+import { useCurrentNodeContext } from 'components/CurrentNodeProvider';
+import { getNodeType } from 'utils/builder';
 
 interface GraphProps {
-  pathway: Pathway;
+  graphContainerWidth: number;
   interactive?: boolean;
-  expandCurrentNode?: boolean;
-  currentNode: PathwayNode;
 }
 
 interface ExpandedNode {
   [key: string]: boolean | string | null;
 }
 
-const Graph: FC<GraphProps> = memo(
-  ({ pathway, interactive = true, expandCurrentNode = true, currentNode }) => {
-    const graphElement = useRef<HTMLDivElement>(null);
-    const nodeRefs = useRef<{ [key: string]: HTMLDivElement }>({});
-    const [parentWidth, setParentWidth] = useState<number>(
-      graphElement?.current?.parentElement?.clientWidth ?? 0
-    );
-
-    // Get the layout of the graph
-    const getGraphLayout = useCallback((): Layout => {
-      const nodeDimensions: NodeDimensions = {};
-
-      // Retrieve dimensions from nodeRefs
-      if (nodeRefs?.current) {
-        Object.keys(nodeRefs.current).forEach(key => {
-          const nodeElement = nodeRefs.current[key];
-          const width = nodeElement.clientWidth;
-          // nodeElement can have multiple children so calculate the sum to get the node height
-          const height = Array.from(nodeElement.children).reduce(
-            (acc, child) => acc + child.clientHeight,
-            0
-          );
-
-          nodeDimensions[key] = { width, height };
-        });
-      }
-
-      return graphLayout(pathway, nodeDimensions);
-    }, [pathway]);
+const Graph: FC<GraphProps> = memo(function Graph({ graphContainerWidth, interactive = true }) {
```

Comment:
Do we need the `function` declaration here? Will this not work as just an arrow function?

Reply:
Yes - without the function wdyr shows this component as `undefined`. This doesn't happen for the other components because they are defined as `const Component: FC = ({}) => {};  export const memo(Component);`. But since this component is memoized here I added a name to the function here